extension/src/openvic-extension/singletons/MapItemSingleton.cpp

```diff
@@ -0,0 +1,229 @@
+
+	return definition_manager->get_map_definition().get_province_definition_count();
+}
+
+PackedVector2Array MapItemSingleton::get_province_positions(){
+	GameSingleton const* game_singleton = GameSingleton::get_singleton();
+	ERR_FAIL_NULL_V(game_singleton, PackedVector2Array());
+
+	DefinitionManager const* definition_manager = &game_singleton->get_definition_manager();
+	ERR_FAIL_NULL_V(definition_manager, PackedVector2Array());
+
+	PackedVector2Array billboard_pos = PackedVector2Array();
+
+	int prov_count = definition_manager->get_map_definition().get_province_definition_count();
+	
+	for(ProvinceDefinition const& prov : definition_manager->get_map_definition().get_province_definitions()){
+		if(prov.is_water()) continue; //billboards dont appear over water, skip
+
+		fvec2_t city_pos = prov.get_city_position();
+		Vector2 pos = Utilities::to_godot_fvec2(city_pos) / game_singleton->get_map_dims();
+		billboard_pos.push_back(pos);
+
+	}
+	
+	return billboard_pos;
+}
+
+//includes non-existent countries, used for setting the billboard buffer size
+int MapItemSingleton::get_capital_count(){
+	GameSingleton const* game_singleton = GameSingleton::get_singleton();
+	ERR_FAIL_NULL_V(game_singleton, 0);
+
+	InstanceManager const* instance_manager = game_singleton->get_instance_manager();
+	ERR_FAIL_NULL_V(instance_manager, 0);
+
+	return instance_manager->get_country_instance_manager().get_country_instance_count();
+}
+
+PackedVector2Array MapItemSingleton::get_capital_positions(){
+	GameSingleton const* game_singleton = GameSingleton::get_singleton();
+	ERR_FAIL_NULL_V(game_singleton, PackedVector2Array());
+
+	InstanceManager const* instance_manager = game_singleton->get_instance_manager();
+	ERR_FAIL_NULL_V(instance_manager, PackedVector2Array());
+
+	PackedVector2Array billboard_pos = PackedVector2Array();
+
+	for(CountryInstance const& country : instance_manager->get_country_instance_manager().get_country_instances()){
+		if(!country.exists()) continue; //skip non-existant countries
+
+		fvec2_t city_pos = country.get_capital()->get_province_definition().get_city_position();
+		Vector2 pos = Utilities::to_godot_fvec2(city_pos) / game_singleton->get_map_dims();
+		billboard_pos.push_back(pos);
+
+	}
+
+	return billboard_pos;
+}
+
+PackedInt32Array MapItemSingleton::get_crime_icons(){
+	GameSingleton const* game_singleton = GameSingleton::get_singleton();
+	ERR_FAIL_NULL_V(game_singleton, PackedInt32Array());
+
+	InstanceManager const* instance_manager = game_singleton->get_instance_manager();
+	ERR_FAIL_NULL_V(instance_manager, PackedInt32Array());
+
+	PackedInt32Array icons = PackedInt32Array();
+
+	for(ProvinceInstance const& prov_inst : instance_manager->get_map_instance().get_province_instances()){
+		ProvinceDefinition const* prov = &prov_inst.get_province_definition();
+		if(prov->is_water()) continue; //billboards dont appear over water, skip
+
+		icons.push_back(prov_inst.get_crime()->get_icon());
```

Comment:
Currently we give each province a crime for demonstration purposes, but in general we can't assume a province's crime isn't `nullpr`. You'll need to check that here and provide a mechanism for not displaying any icon in that case. The same applies for RGOs, I'm fairly sure it's possible for a province to have no RGO if one isn't given in its history file.

If you can make it so we can selectively hide specific province's icons without needing to re-send the whole list of positions, it would be useful to handle capitals in a similar way: rather than sending a list of all current capitals, we could use a list of all province city positions (this list of positions would be the same for capitals and RGO/crime icons) and then update it with a list of bools indicating whether each province is a capital or not.

Reply:
We decided that having one capital per province wouldn't be efficient. Positions don't need to be re-sent currently with the exception of capitals.